tests/data/test_dataset_generator.py

```diff
@@ -0,0 +1,150 @@
+"""Test the DatasetGenerator classes."""
+
+from unittest.mock import Mock
+
+import numpy as np
+import pandas as pd
+import pytest
+from pytest_mock import MockerFixture
+
+from coin_test.data import ReturnsDatasetGenerator
+from coin_test.data.metadata import MetaData
+from coin_test.util import AssetPair, Ticker
+
+
+def test_dataset_generator_initialized(hour_data_indexed_df: pd.DataFrame) -> None:
+    """Initialize the ResultDatasetGenerator."""
+    mock_dataset = Mock()
+    pair = AssetPair(Ticker("BTC"), Ticker("USDT"))
+    metadata = MetaData(pair, freq="H")
+    mock_dataset.metadata = metadata
+    mock_dataset.df = hour_data_indexed_df
+
+    gen = ReturnsDatasetGenerator(mock_dataset)
+
+    assert gen.dataset == mock_dataset
+    assert gen.metadata == metadata
+    assert isinstance(gen.start, pd.Period)
+
+
+def test_normalize_dataset(
+    hour_data_indexed_df: pd.DataFrame, hour_data_norm_df: pd.DataFrame
+) -> None:
+    """Normalize data with ResultsDatasetGenerator."""
+    norm_df = ReturnsDatasetGenerator.normalize_row_data(hour_data_indexed_df)
+
+    pd.testing.assert_frame_equal(norm_df, hour_data_norm_df)
+
+    # with open("tests/data/assets/norm_eth_usdc_1h_9_28.csv", "w") as outfile:
+    #     norm_df.to_csv(outfile, index=False)
+
+
+def test_dataset_select_data(
+    hour_data_norm_df: pd.DataFrame, hour_data_sel_df: pd.DataFrame
+) -> None:
+    """Select data with ResultsDatasetGenerator."""
+    rng = np.random.default_rng(int("stonks", 36))
+    starting_price = 7.48
+    num_rows = 4
+
+    selected_df = ReturnsDatasetGenerator.select_data(
+        hour_data_norm_df, starting_price, num_rows, rng
+    )
+
+    pd.testing.assert_frame_equal(selected_df, hour_data_sel_df)
```

Comment:
Were the dfs that are being loaded checked for correctness by hand? Would like to avoid checking the algorithm with output from the same algorithm if possible, but if we know the saved output is valid then it seems safe.